### DISCUSSION TITLE
3D model export

### DISCUSSION THREAD

**Discussion by MokosDaniel** [BradyAJohnston/MolecularNodes#340] [Q&A] [answered]
Does anyone have a way to export the models as 3D objects? I am working on augmented reality, and I would like to get a .gltf file exported from blender, however any time I export I get an empty file.
A stationary object would do, however an animated model would be the best if possible.

**Comment by BradyAJohnston** [comment]
I haven't done a large amount of exporting as 3D models, but you should be able to. You just have to be conscious of the styles that are being used. The 'Atom' & 'ball-and-stick' styles might not export well, as they are using point-clouds and instancing which are likely not compatible with a bunch of different export methods.

The surface / cartoon / ribbon should all export OK though. Is there an example model that you are interested in exporting? I can test out some approaches on it

**Reply by MokosDaniel** [reply]
Thank you for your response, I tried exporting surfaces now, and that does work for single frames, however I was unable to get the animation applied to the realized mesh, and to the exported model. Do you have any suggestions for that? I'll admit I am not a very experienced blender user...
I would like to be able to do it in general currently, with for example a multi-state pdb, with the automatic interpolation you already implemented nicely. I am currently using a few frames of an MD sim I received from a colab-partner, and combined them into a single PDB as consecutive states.
I am mostly interested in mechanisms, so ball-and-stick would be the best at least for ligends/sidechains, however I can work with other options as well.

**Reply by BradyAJohnston** [reply, marked as answer]
I've had a bit of a play around, and wasn't able to get the animation to export either. I think it's a limitation of the way Molecular Nodes works in terms of using a geometry modifier to animation the mesh, rather than traditional bones etc that are usually used to animate meshes for 3D formats.

There would be potential to maybe convert the molecular nodes setup to a bone-controlled animation, but this would likely require a fair bit of dev work and wouldn't be a priority of mine at the moment. 

More control over bones / rigging / animation in general is something that I have been thinking about for a while, and there have been a few requests for exporting of animations, so it will likely be a project at some point but I wouldn't say it's currently of a high priority.

**Reply by MokosDaniel** [reply]
Thanks for your answer, keep up the good work!
